File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        var stack = new Stack<int>();
        foreach (var token in tokens) {
            if (token == "+") {
                int a = stack.Pop();
                int b = stack.Pop();
                stack.Push(a + b);
            } else if (token == "-") {
                int a = stack.Pop();
                int b = stack.Pop();
                stack.Push(b - a);
            } else if (token == "*") {
                int a = stack.Pop();
                int b = stack.Pop();
                stack.Push(a * b);
            } else if (token == "/") {
                int a = stack.Pop();
                int b = stack.Pop();
                stack.Push(b / a);
            } else {
                stack.Push(int.Parse(token));
            }
        }

        return stack.Pop();
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-2.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        out = [0] * len(temperatures)
        for i, t in enumerate(temperatures):
            while stack:
                if temperatures[stack[-1]] < t:
                    j = stack.pop()
                    out[j] = i - j
                else:
                    break
            stack.append(i)
        return out